tests/test_core.py

```diff
@@ -700,3 +701,48 @@ def test_import_image(self, openshift, tags, imagestream_name, expect_update, ex
             .with_args(post_url, data=JsonMatcher(stream_import_json), use_json=True))
 
         assert openshift.import_image(imagestream_name, stream_import, tags=tags) is expect_import
+
+    @pytest.mark.parametrize(('image_status', 'expect_retry'), (  # noqa:F811
+        ({'status': 'Failure', 'code': 500, 'reason': 'InternalError'}, True),
+        ({'status': 'Failure', 'code': 400, 'reason': 'BadRequest'}, False),
+    ))
+    def test_import_image_retry(self, openshift, image_status, expect_retry):
+        imagestream_name = TEST_IMAGESTREAM
+
+        # Load example API response
+        this_file = inspect.getfile(TestCheckResponse)
+        this_dir = os.path.dirname(this_file)
+        json_path = os.path.join(this_dir, "mock_jsons", openshift._con.version,
+                                 'imagestreamimport.json')
+        with open(json_path) as f:
+            content_json = json.load(f)
+
+        # Assume mocked data contains a good response
+        good_resp = HttpResponse(200, {}, content=json.dumps(content_json).encode('utf-8'))
+
+        # Create a bad response by marking the first image as failed
+        for key in ('status', 'code', 'reason'):
+            content_json['status']['images'][0]['status'][key] = image_status[key]
```

Comment:
Now that I look at this, I guess it could be:
```
content_json['status']['images'][0]['status'].update(image_status)
```
but that's just being picky now.

Reply:
That's true, but I wanted to be more explicit in case someone added a new use case in this unit test that had a typo in the keys used.